lib/awful/tag.lua

```diff
@@ -749,6 +766,32 @@ function tag.object.set_layout(t, layout)
     return layout
 end
 
+function tag.object.get_layouts(self)
+    -- The override has the priority.
+    local cls = tag.getproperty(self, "custom_layouts")
+
+    -- Allow custom layouts to be listed until GC-ed.
+    if not cls then
+        cls = setmetatable({}, {__mode="v"})
+        tag.setproperty(self, "custom_layouts", cls)
```

Comment:
Why this GC magic? Can't you just return the table 'normally'? Do any callers save the table? If so, why is it fine if the GC "silently messes" with the table?

Reply:
Because layouts can be objects (that part of my dynamic work was merged before 4.0) and it will leak if it uses a normal table.

Reply:
Huh? What would it leak? If it leaks a layout, then I still have to ask again: "why is it okay when the list of available layouts on a tag changes randomly" (where "randomly" depends on the behaviour of the GC)